clients/src/main/java/org/apache/kafka/common/requests/ResponseHeader.java

```diff
@@ -16,49 +16,57 @@
  */
 package org.apache.kafka.common.requests;
 
-import org.apache.kafka.common.protocol.types.BoundField;
-import org.apache.kafka.common.protocol.types.Field;
-import org.apache.kafka.common.protocol.types.Schema;
+import org.apache.kafka.common.message.ResponseHeaderData;
+import org.apache.kafka.common.protocol.ByteBufferAccessor;
 import org.apache.kafka.common.protocol.types.Struct;
 
 import java.nio.ByteBuffer;
 
-import static org.apache.kafka.common.protocol.types.Type.INT32;
-
 /**
  * A response header in the kafka protocol.
  */
 public class ResponseHeader extends AbstractRequestResponse {
-    public static final Schema SCHEMA = new Schema(
-            new Field("correlation_id", INT32, "The user-supplied value passed in with the request"));
-    private static final BoundField CORRELATION_KEY_FIELD = SCHEMA.get("correlation_id");
+    private final ResponseHeaderData data;
+    private final short headerVersion;
 
-    private final int correlationId;
+    public ResponseHeader(Struct struct, short headerVersion) {
+        this.data = new ResponseHeaderData(struct, headerVersion);
```

Comment:
nit: chain these c'tors to consolidate code. Makes it easy to do validation etc in case a need arise in future.